packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/include/video_player_avfoundation/FVPVideoPlayer.h

```diff
@@ -0,0 +1,59 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#if TARGET_OS_OSX
+#import <FlutterMacOS/FlutterMacOS.h>
+#else
+#import <Flutter/Flutter.h>
+#endif
+
+#import <AVFoundation/AVFoundation.h>
+#import <GLKit/GLKit.h>
+
+#import "FVPVideoPlayerPlugin_Test.h"
+
+#pragma mark -
+
+@interface FVPVideoPlayer ()
+@property(readonly, nonatomic) AVPlayerItemVideoOutput *videoOutput;
+// The plugin registrar, to obtain view information from.
+@property(nonatomic, weak) NSObject<FlutterPluginRegistrar> *registrar;
+@property(nonatomic) FlutterEventChannel *eventChannel;
+@property(nonatomic) FlutterEventSink eventSink;
+@property(nonatomic) CGAffineTransform preferredTransform;
+@property(nonatomic) AVPlayerLayer *playerLayer;
+@property(nonatomic, readonly) BOOL disposed;
+@property(nonatomic, readonly) BOOL isPlaying;
+@property(nonatomic) BOOL isLooping;
+@property(nonatomic, readonly) BOOL isInitialized;
+
+- (instancetype)initWithURL:(NSURL *)url
+                httpHeaders:(nonnull NSDictionary<NSString *, NSString *> *)headers
+                  avFactory:(id<FVPAVFactory>)avFactory
+                  registrar:(NSObject<FlutterPluginRegistrar> *)registrar;
+
+- (instancetype)initWithPlayerItem:(AVPlayerItem *)item
+                         avFactory:(id<FVPAVFactory>)avFactory
+                         registrar:(NSObject<FlutterPluginRegistrar> *)registrar;
+
+- (instancetype)initWithAsset:(AVPlayerItem *)item
+                    avFactory:(id<FVPAVFactory>)avFactory
+                    registrar:(NSObject<FlutterPluginRegistrar> *)registrar;
+
+- (void)updatePlayingState;
+
+- (void)dispose;
+
+- (void)disposeSansEventChannel;
+
+- (void)setVolume:(double)volume;
+
+- (void)setPlaybackSpeed:(double)speed;
+
+- (void)play;
+
+- (void)pause;
+
+- (void)seekTo:(int64_t)location completionHandler:(void (^)(BOOL))completionHandler;
```

Comment:
I had to add many of these methods to the interface because they were referenced in `FVPVideoPlayerPlugin.m` (and now the player is not longer a part of that file)